app/policies/application_policy.rb

```diff
@@ -39,6 +39,10 @@ def user_owns_record?
     record.user_id == user.id
   end
 
+  def volunteer_related?
+    record.volunteer.user.id == user.id
```

Comment:
Slightly better: `record.volunteer.user_id == user.id` (avoids loading the `user` relation if it's not loaded yet)